beacon_node/beacon_chain/src/data_column_verification.rs

```diff
@@ -221,34 +222,91 @@ impl<E: EthSpec> KzgVerifiedDataColumn<E> {
         self.data.clone()
     }
 
-    pub fn data_column_index(&self) -> u64 {
+    pub fn index(&self) -> ColumnIndex {
         self.data.index
     }
 }
 
-/// Data column that we must custody and has completed kzg verification
-#[derive(Debug, Derivative, Clone, Encode, Decode)]
-#[derivative(PartialEq, Eq)]
-#[ssz(struct_behaviour = "transparent")]
-pub struct KzgVerifiedCustodyDataColumn<E: EthSpec> {
-    data: Arc<DataColumnSidecar<E>>,
+/// Collection of data columns for the same block root
+pub struct DataColumnsSameBlock<E: EthSpec> {
+    block_root: Hash256,
+    signed_block_header: SignedBeaconBlockHeader,
+    columns: DataColumnSidecarList<E>,
 }
 
-impl<E: EthSpec> KzgVerifiedCustodyDataColumn<E> {
-    /// Mark a column as custody column. Caller must ensure that our current custody requirements
-    /// include this column
-    pub fn from_asserted_custody(kzg_verified: KzgVerifiedDataColumn<E>) -> Self {
-        Self {
-            data: kzg_verified.to_data_column(),
+impl<E: EthSpec> DataColumnsSameBlock<E> {
+    pub fn new(columns: DataColumnSidecarList<E>) -> Result<Self, &'static str> {
+        let first_column = columns.first().ok_or("empty columns")?;
+        let signed_block_header = first_column.signed_block_header.clone();
+        for column in columns.iter().skip(1) {
+            if column.signed_block_header != signed_block_header {
+                return Err("no same block");
```

Comment:
`not same block` or maybe more detailed`mismatched data column headers` or something